controller/internal/routes/terminator_api_model.go

```diff
@@ -0,0 +1,131 @@
+/*
+	Copyright 2020 NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package routes
+
+import (
+	"fmt"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/netfoundry/ziti-edge/controller/env"
+	"github.com/netfoundry/ziti-edge/controller/response"
+	"github.com/netfoundry/ziti-fabric/controller/models"
+	"github.com/netfoundry/ziti-fabric/controller/network"
+	"github.com/netfoundry/ziti-foundation/util/stringz"
+)
+
+const EntityNameTerminator = "terminators"
+
+type TerminatorApi struct {
+	Service *string `json:"service"`
+	Router  *string `json:"router"`
+	Binding *string `json:"binding"`
+	Address *string `json:"address"`
+}
+
+func (i *TerminatorApi) ToModel(id string) *network.Terminator {
+	result := &network.Terminator{}
+	result.Id = id
+	result.Service = stringz.OrEmpty(i.Service)
+	result.Router = stringz.OrEmpty(i.Router)
+	result.Binding = stringz.OrEmpty(i.Binding)
+	result.Address = stringz.OrEmpty(i.Address)
+
+	return result
+}
+
+type TerminatorApiList struct {
+	*env.BaseApi
+	ServiceId string        `json:"serviceId"`
+	Service   *EntityApiRef `json:"service"`
```

Comment:
entityRefs contain a property called `id`

I am having trouble convincing myself that having both is useful.

Reply:
yes, I was having the same internal debate. I was following the model you posted in slack the other day. Should I remove the *Id fields?